File: app/src/test/java/com/farmerbb/taskbar/util/TopAppsTest.java
package com.farmerbb.taskbar.util;

import android.content.Context;

import androidx.test.core.app.ApplicationProvider;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

@RunWith(RobolectricTestRunner.class)
public class TopAppsTest {
    private Context context;
    private TopApps topApps;

    @Before
    public void setUp() {
        context = ApplicationProvider.getApplicationContext();
        topApps = TopApps.getInstance(context);
    }

    @Test
    public void testGetInstance() {
        assertNotNull(topApps);
        for (int i = 1; i <= 20; i++) {
            assertEquals(topApps, TopApps.getInstance(context));
        }
    }

    @Test
    public void testIsTopApp() {
        assertFalse(topApps.isTopApp(context.getPackageName()));
        addCurrentPackageToTopApps();
        assertTrue(topApps.isTopApp(context.getPackageName()));
        topApps.removeTopApp(context, context.getPackageName());
    }

    @Test
    public void testRemoveTopApp() {
        addCurrentPackageToTopApps();
        assertTrue(topApps.isTopApp(context.getPackageName()));
        topApps.removeTopApp(context, context.getPackageName());
        assertFalse(topApps.isTopApp(context.getPackageName()));
    }

    private void addCurrentPackageToTopApps() {
        String packageName = context.getPackageName();
        BlacklistEntry entry = new BlacklistEntry(packageName, packageName);
        topApps.addTopApp(context, entry);
    }
}

File: app/src/test/java/com/farmerbb/taskbar/util/TopAppsTest.kt
package com.farmerbb.taskbar.util

import android.content.Context
import androidx.test.core.app.ApplicationProvider
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner

@RunWith(RobolectricTestRunner::class)
class TopAppsTest {
    private lateinit var context: Context
    private lateinit var topApps: TopApps

    @Before
    fun setUp() {
        context = ApplicationProvider.getApplicationContext()
        topApps = TopApps.getInstance(context)
    }

    @Test
    fun testGetInstance() {
        Assert.assertNotNull(topApps)
        for (i in 1..20) {
            Assert.assertEquals(topApps, TopApps.getInstance(context))
        }
    }

    @Test
    fun testIsTopApp() {
        Assert.assertFalse(topApps.isTopApp(context.packageName))
        addCurrentPackageToTopApps()
        Assert.assertTrue(topApps.isTopApp(context.packageName))
        topApps.removeTopApp(context, context.packageName)
    }

    @Test
    fun testRemoveTopApp() {
        addCurrentPackageToTopApps()
        Assert.assertTrue(topApps.isTopApp(context.packageName))
        topApps.removeTopApp(context, context.packageName)
        Assert.assertFalse(topApps.isTopApp(context.packageName))
    }

    private fun addCurrentPackageToTopApps() {
        val packageName = context.packageName
        val entry = BlacklistEntry(packageName, packageName)
        topApps.addTopApp(context, entry)
    }
}
